### PR TITLE
adding base java client with listMetrics

### PR DESCRIPTION
**Pull request by Geo881**
### Summary

adding base java client with list metrics

### Test Plan

- [x] PR has an associated issue: #502
- [x] `make check` passes
- [x] `make test` shows 100% unit test coverage

### Deployment Plan

n/a

### CODE REVIEW

**Comment by agorajek** [comment]
Hey @Geo881 thanks for starting the journey on the Java client! I think its totally fine to add things slowly, but would you also mind to start with a unit test harness and the first unit test for your `listMetrics()` ?

**Review by samredai — APPROVED** [review]
Thanks @Geo881! I'll go ahead and merge this. Some nice follow-ups would be to get these tests ran as part of the client integration GitHub action and also to add a [google-java-format](https://github.com/google/google-java-format) linter just to keep things consistent going forward.

File: out/production/dj/io/datajunction/client/DJClient.class
Comment (agorajek) [inline out/production/dj/io/datajunction/client/DJClient.class:1]: @samredai @Geo881 I just noticed this top level dir/file in the repo and I am wondering if this is some sort of Java. standard or is it here by accident?

Comment (agorajek) [reply out/production/dj/io/datajunction/client/DJClient.class:1]: @samredai @Geo881 pinging again on my question...

**Pull request merged**